crates/astria-sequencer/src/mempool/transactions_container.rs

```diff
@@ -189,6 +287,19 @@ impl TransactionsForAccount for PendingTransactionsForAccount {
         // is equal to the account nonce
         self.txs().contains_key(&previous_nonce) || ttx.signed_tx.nonce() == current_account_nonce
     }
+
+    fn has_balance_to_cover(
+        &self,
+        ttx: &TimemarkedTransaction,
+        current_account_balances: &HashMap<IbcPrefixed, u128>,
+    ) -> bool {
+        let mut current_account_balances = current_account_balances.clone();
```

Comment:
This is a pretty innocent looking functions where one does not expect a deep copy to happen.

Reply:
Ah, good catch. Will change!

Reply:
Actually, this was done to prevent doing copies of the `account_balances` for the parked transactions container, which is a noop for the `has_balance_to_cover` fn. We can avoid making unnecessary copies if we do the clone on the inside for the pending container